docs/static/troubleshooting.asciidoc

```diff
@@ -0,0 +1,270 @@
+
+*Symptoms*
+
+Logstash’s Kafka Input is causing a much higher number of commits to
+the offset topic than expected. Often the complaint also mentions redundant
+offset commits where the same offset is committed repeatedly.
+
+*Solution*
+
+For Kafka Broker versions 0.10.2.1 to 1.0.x: The problem is caused by a bug in
+Kafka. https://issues.apache.org/jira/browse/KAFKA-6362 The client’s best option
+is upgrading their Kafka Brokers to version 1.1 or newer. 
+
+For older versions of
+Kafka or if the above does not fully resolve the issue: The problem can also be
+caused by setting the value for `poll_timeout_ms` too low relative to the rate
+at which the Kafka Brokers receive events themselves (or if Brokers periodically
+idle between receiving bursts of events). Increasing the value set for
+`poll_timeout_ms` proportionally decreases the number of offsets commits in
+this scenario. For example, raising it by 10x will lead to 10x fewer offset commits.
+
+
+[float] 
+[[ts-kafka-codec-errors-input]] 
+=== Codec Errors in Kafka Input (before Plugin Version 6.3.4 only) 
+
+*Symptoms*
+
+Logstash Kafka input randomly logs errors from the configured codec and/or reads
+events incorrectly (partial reads, mixing data between multiple events etc.).
+
+-----
+Log example:  [2018-02-05T13:51:25,773][FATAL][logstash.runner          ] An
+unexpected error occurred! {:error=>#<TypeError: can't convert nil into String>,
+:backtrace=>["org/jruby/RubyArray.java:1892:in `join'",
+"org/jruby/RubyArray.java:1898:in `join'",
+"/usr/share/logstash/logstash-core/lib/logstash/util/buftok.rb:87:in `extract'",
+"/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-line-3.0.8/lib/logstash/codecs/line.rb:38:in
+`decode'",
+"/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-kafka-5.1.11/lib/logstash/inputs/kafka.rb:241:in
+`thread_runner'",
+"file:/usr/share/logstash/vendor/jruby/lib/jruby.jar!/jruby/java/java_ext/java.lang.rb:12:in
+`each'",
+"/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-kafka-5.1.11/lib/logstash/inputs/kafka.rb:240:in
+`thread_runner'"]} 
+-----
+
+*Background*
+
+There was a bug in the way the Kafka Input plugin was handling codec instances
+when running on multiple threads (`consumer_threads` set to > 1).
+https://github.com/logstash-plugins/logstash-input-kafka/issues/210 
+
+*Solution*
+
+* Upgrade Kafka Input plugin to v. 6.3.4 or later. 
+* If (and only if) upgrading is impossible, set `consumer_threads` to `1`.
+
+
+[float] 
+[[ts-other]] 
+== Other issues
+
+[float] 
+[[ts-cli]] 
+=== Command line
+
+[float] 
+[[ts-windows-cli]] 
+==== Shell commands on Windows OS
+
+Command line often show single quotes. 
+On Windows systems, replace a single quote `'' with a double quote `"`. 
+
+*Example*
+
+Instead of:
+
+-----
+bin/logstash -e 'input { stdin { } } output { stdout {} }'
+-----
+
+Use this format on Windows systems:
+
+-----
+bin/logstash -e "input { stdin { } } output { stdout {} }"
```

Comment:
also you need to invert that forward slash: `bin\logstash` as paths in windows are separated by backslash